Assets/_Project/Code/Scripts/Bidding_War/Bidding_War_Controller.cs

```diff
@@ -0,0 +1,261 @@
+            Total_Bid_Text.text = "aaaaa";
+        }
+        int i = Teams.Count;
+        while (i < Team_Names_Text.Count)
+        {
+            Destroy(Team_Names_Text[i]);
+            Destroy(Team_Bid_Text[i]);
+            Destroy(Team_Balance_Text[i]);
+            i += 1;
+        }
+        Setup(); ;
+        Add_Listners(); 
+    }
+
+
+    
+    public void Add_Listners() 
+    {
+        Bid_Buttons[0].onClick.AddListener(delegate { Bid(100); });
+        Bid_Buttons[1].onClick.AddListener(delegate { Bid(200); });
+        Bid_Buttons[2].onClick.AddListener(delegate { Bid(300); });
+        Bid_Buttons[3].onClick.AddListener(delegate { Bid(400); });
+        Bid_Buttons[4].onClick.AddListener(delegate { Bid(500); });
+        Bid_Buttons[5].onClick.AddListener(delegate { Bid(1000); });
+        VB_Button.onClick.AddListener(delegate { Va_Banque(); });
+        ExitButton.onClick.AddListener(delegate { Exit_To_Lobby(); });
+    }
+   
+    void Setup()
+    {
+       int i = 0;
+        while (i < Teams.Count) 
+        { 
+            Team_Balance_Text[i].text = Teams[i].Money.ToString();
+            Team_Bid_Text[i].text = Teams[i].Bid.ToString();
+            Team_Names_Text[i].text = Teams[i].Colour;
+            i += 1;
+        }
+        Reset_Timer();
+    }
+
+    [Rpc(SendTo.Everyone)]
+    void Setup_Stage_2_Rpc()
+    {
+        int i = 0;
+        while (i < Teams.Count) 
+        {
+            Teams[i].Raise_Bid(500);
+            Update_Money_Status_For_Team(i);
+            i += 1;
+            _winning_bid_amount = 500;
+            _total_bid = Teams.Count * 500;
+            Total_Bid_Text.text = _total_bid.ToString();
+        }
+        _game_ongoing = true;
+    }
+
+    public void Update_Money_Status_For_Team(int i) 
+    {
+        Team_Balance_Text[i].text = Teams[i].Money.ToString();
+        Team_Bid_Text[i].text = Teams[i].Bid.ToString();
+    }
+
+    public void Update_Money_Status()
+    {
+        int i = 0;
+        while (i < Teams.Count) 
+        {
+            Update_Money_Status_For_Team(i);
+            i += 1;  
+        }
+        Update_Buttons();
+        Total_Bid_Text.text = _total_bid.ToString();
+    }
+
+    public void Update_Buttons() 
+    {
+        if (_winning_bid_amount != Teams[(int)_player_id].Bid)
+        {
+            int difference = _winning_bid_amount - Teams[(int)_player_id].Bid;
+
+            Bid_Button_Text[0].text = "100(" + (difference + 100).ToString() + ")";
+            Bid_Button_Text[1].text = "200(" + (difference + 200).ToString() + ")";
+            Bid_Button_Text[2].text = "300(" + (difference + 300).ToString() + ")";
+            Bid_Button_Text[3].text = "400(" + (difference + 400).ToString() + ")";
+            Bid_Button_Text[4].text = "500(" + (difference + 500).ToString() + ")";
+            Bid_Button_Text[5].text = "1000(" + (difference + 1000).ToString() + ")";
+        }
```

Comment:
zrób sobie zmienną bidding options i wtedy mozesz dynamicznie tworzyć guzik w zależności od ich ilości

Reply:
w sumie to na razie nie rób pętli, bo przekomplikowane może być. Ale tego ifa poteżnego jakoś uprość, bo sporo logiki się powtarza